pkg/aws/region/flag.go

```diff
@@ -24,18 +24,29 @@ import (
 	"github.com/spf13/pflag"
 
 	"github.com/openshift/rosa/pkg/helper"
+	"github.com/openshift/rosa/pkg/reporter"
 )
 
-// AddFlag adds the debug flag to the given set of command line flags.
+var regionFlag = "region"
+
+// AddFlag adds the region flag to the given set of command line flags.
 func AddFlag(flags *pflag.FlagSet) {
 	flags.StringVar(
 		&region,
-		"region",
+		regionFlag,
 		"",
 		"Use a specific AWS region, overriding the AWS_REGION environment variable.",
 	)
 }
 
+// DisableFlag 'disables' the region flag from being used in the given set of command line flags.
+// This command checks if the region flag is used when it shouldn't be, then errors and exits if it is
+func DisableFlag(flags *pflag.FlagSet, reporter *reporter.Object) {
```

Comment:
@hunterkepley I think it would be useful for this to be shown in the `help` output for a command as well. To achieve that, we need the following for these commands when their defining their flagset:

```
cmd.Flags().MarkDeprecated("region", "Region flag will be removed from this command in future versions")
cmd.Flag("region").Hidden = false
```

So we could wrap that into a function that gets called when creating the flags for a command e.g:

```
func MarkRegionDeprecated(cmd *cobra.Command) {
flag := cmd.Flag("region")
if flag != nil {
  cmd.Flags().MarkDeprecated("cluster", "Region flag will be removed from this command in future versions")
  cmd.Flag("cluster").Hidden = false
  }
}
```

That seems to cause a warning to be printed when the user makes use of the deprecated flag, which would drop the need for the per-command log statement.